ompi/mpi/fortran/use-mpi-f08/mpi-f-interfaces-bind.h

```diff
@@ -935,10 +946,11 @@ subroutine ompi_ialltoallw_f(sendbuf,sendcounts,sdispls,sendtypes, &
                             recvbuf,recvcounts,rdispls,recvtypes,comm,request,ierror) &
    BIND(C, name="ompi_ialltoallw_f")
    implicit none
-   OMPI_FORTRAN_IGNORE_TKR_TYPE, INTENT(IN) :: sendbuf, recvbuf
-   INTEGER, INTENT(IN) :: sendcounts(*), sdispls(*), recvcounts(*), rdispls(*)
-   INTEGER, INTENT(IN) :: sendtypes
-   INTEGER, INTENT(IN) :: recvtypes
+   OMPI_FORTRAN_IGNORE_TKR_TYPE, INTENT(IN), ASYNCHRONOUS :: sendbuf
+   OMPI_FORTRAN_IGNORE_TKR_TYPE, ASYNCHRONOUS :: recvbuf
+   INTEGER, INTENT(IN), ASYNCHRONOUS :: sendcounts(*), sdispls(*), recvcounts(*), rdispls(*)
+   INTEGER, INTENT(IN), ASYNCHRONOUS :: sendtypes(*)
+   INTEGER, INTENT(IN), ASYNCHRONOUS :: recvtypes(*)
```

Comment:
You changed the types of `sendtypes` and `recvtypes` from scalars to arrays here.  They were specifically scalars (in both `ompi_alltoallw_f` and `ompi_ialltoallw_f`) -- see the lengthy comment here: https://github.com/open-mpi/ompi/blob/master/ompi/mpi/fortran/use-mpi-f08/ialltoallw_f08.F90#L24-L33

Yes, it's a hack, but there did not seem to be another performant way to get the values from one call to the other (i.e., without allocating a temporary array from the heap [because this is ialltoallw, and we'll return before sendtypes/recvtypes are finished being used], individually copying all the elements from sendtypes/recvtypes to the temporary array, and then passing the temporary array to the underlying call.  Then there will need to be some kind of hook when the request completes to know to free that temporary array, too).

Is there a better way?

Reply:
Fortran allows you to obtain the address of the first element in an array via the intrinsic c_loc which returns a type(c_ptr) which is equivalent to a void pointer in C. There would be no copying involved and you would prevent the compiler from doing any unwanted magic there.

Reply:
Can you add that to this PR?  I'm wasn't enough of a Fortran expert to get that to work back when I originally wrote this stuff -- this hack was the best that I could come up with.  😳

Reply:
According to Modern Fortran explained by Metcalf, Reid and Cohen the proposed way of using c_loc may lead to invalid pointers. So, your hack is probably the safer way of doing things.